src/frontend/sharedComponents/ActionButtons.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import {Alert, StyleSheet, TouchableOpacity, View} from 'react-native';
+import {UIActivityIndicator} from 'react-native-indicators';
+import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
+import {Text} from './Text';
+import {DARK_GREY, LIGHT_GREY} from '../lib/styles';
+import {MessageDescriptor, defineMessages, useIntl} from 'react-intl';
+
+const m = defineMessages({
+  delete: {
+    id: 'SharedComponents.ActionButtons.delete',
+    defaultMessage: 'Delete',
+    description: 'Button to delete an observation',
+  },
+  share: {
+    id: 'SharedComponents.ActionButtons.share',
+    defaultMessage: 'Share',
+    description: 'Button to share an observation',
+  },
+  cancel: {
+    id: 'SharedComponents.ActionButtons.cancel',
+    defaultMessage: 'Cancel',
+    description: 'Button to cancel delete of observation',
+  },
+  confirm: {
+    id: 'SharedComponents.ActionButtons.confirm',
+    defaultMessage: 'Yes, delete',
+    description: 'Button to confirm delete of observation',
+  },
+});
+
+type ActionButtonsProps = {
+  handleDelete: () => void;
+  handlePressShare?: () => void;
+  isMine: boolean;
+  isShareButtonLoading?: boolean;
+  deleteMessage: MessageDescriptor;
+};
+
+export const ActionButtons = ({
+  handleDelete,
+  handlePressShare,
+  isMine,
+  isShareButtonLoading,
+  deleteMessage,
+}: ActionButtonsProps) => {
+  const {formatMessage: t} = useIntl();
+
+  function pressDeleteAlert() {
+    Alert.alert(t(deleteMessage), undefined, [
+      {
+        text: t(m.cancel),
+        onPress: () => {},
+      },
+      {
+        text: t(m.confirm),
+        onPress: handleDelete,
+      },
+    ]);
+  }
```

Comment:
outside the scope of this PR since this was here before, but shouldn't this be using a modal and not an alert? 😅